### DISCUSSION TITLE
Database Notifications stored with Unix Timestamp

### DISCUSSION THREAD

**Discussion by rileysiemens** [laravel/framework#48720] [Q&A] [answered]
Hey everyone!

The project I'm working on stores all timestamps as Unix timestamps.  We also us a custom wrapper around Carbon (that we call "Nitrogen") that does some date & translation specific things for our project.  All of our timestamps are stored as Unix timestamps and are cast using our custom class.  

Our `app\Models\Model.php` looks like this:
```php
<?php

namespace App\Models;

use App\Casts\NitrogenCast;
use \Illuminate\Database\Eloquent\Model as EloquentModel;

class Model extends EloquentModel {
    protected $dateFormat = 'U';
    protected $casts = [
        'created_at' => NitrogenCast::class,
        'updated_at' => NitrogenCast::class,
    ];
}
```

I'm implementing notifications in my project, and I'd like to do similar things with DatabaseNotifications stored in the "notifications" table.  Any ideas how I could do that?

**Comment by iazaran** [comment]
I think making a model that extends DatabaseNotification _src/Illuminate/Notifications/DatabaseNotification.php_ like `class NitrogenDatabaseNotification extends DatabaseNotification` can help us to add `$dateFormat` and override `$casts`.

**Reply by rileysiemens** [reply]
Thanks for the idea. 

I made a custom class and named it "Database Notification" thinking that maybe the framework would pick up on a user-defined model with the same name and use it instead of the default.

```php
<?php

namespace App\Models;

use App\Casts\NitrogenCast;
use \Illuminate\Notifications\DatabaseNotification as LaravelDatabaseNotification;

class DatabaseNotification extends LaravelDatabaseNotification {
    protected $dateFormat = 'U';
    protected $casts = [
        'created_at' => NitrogenCast::class,
        'updated_at' => NitrogenCast::class,
    ];
}
```
But this didn't work.  It's still trying to store the timestamps as strings (which doesn't work because in the migration I created the columns as integers).

Is there anyway to "publish" the file so that you can force Laravel to use my custom class instead?

**Reply by iazaran** [reply, marked as answer]
Because of using Notifiable trait in models like User model, I think that trait needs to be extended or overriding to consider your `DatabaseNotification` model. And then we can use that trait instead of default one.
I didn't test this scenario, but it is good idea if you check the _src/Illuminate/Notifications/HasDatabaseNotifications.php_ you can see the trait is using default `DatabaseNotification` and so it can be overridden.

**Reply by rileysiemens** [reply]
Ah yes, good idea. Thank you! I'll try that and see how it goes.

**Reply by rileysiemens** [reply]
Update: This worked like a charm!  Thanks!
